app/src/main/java/ch/epfl/culturequest/ArtDescriptionDisplayActivity.java

```diff
@@ -108,6 +130,36 @@ private void setTextOrFallback(TextView textView, String text, String fallbackTe
         textView.setText(text != null ? text : fallbackText);
     }
 
+    private void showOpenAIPopup() {
+        View popupView = LayoutInflater.from(this).inflate(R.layout.popup_openai_message, null);
+        final PopupWindow popupWindow = new PopupWindow(popupView, ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT, true);
+
+        // Show the popup in the center of the screen after the activity has been fully initialized
+        findViewById(android.R.id.content).post(() -> {
+            if (!isFinishing()) {
+                popupWindow.showAtLocation(findViewById(android.R.id.content), Gravity.CENTER, 0, 0);
+            }
+        });
+
+        // Set the "Do not show it again" button click listener
+        Button btnDoNotShowAgain = popupView.findViewById(R.id.btn_do_not_show_again);
+        btnDoNotShowAgain.setOnClickListener(view -> {
+
+            SharedPreferences sharedPreferences = getSharedPreferences("openAI_popup_pref", MODE_PRIVATE);
+            SharedPreferences.Editor editor = sharedPreferences.edit();
+            editor.putBoolean("do_not_show_again", true);
+            editor.apply();
+
+            popupWindow.dismiss();
+        });
+
+        // Set a delay of 5 seconds before making the popup invisible
+        new Handler().postDelayed(() -> {
+            if (popupWindow.isShowing()) {
+                popupWindow.dismiss();
+            }
+        }, 5000);
+    }
```

Comment:
The 5 seconds are purely arbitrary. I took into account the fact that clicking on the "Don't show again" button would immediately hide the popup, so there's actually a way to make it last a shorter amount of time.
If you think that it's too long, I can reduce it to 3 seconds or something like that.

Could you clarify your second sentence? As of now, the warning popup is shown right after the description activity is displayed, so that the code can know that the description required OpenAI.

Reply:
If it is hardcoded then you should create a constant for it